client/consul/sync.go

```diff
@@ -351,31 +380,54 @@ func (c *ConsulService) deregisterCheck(ID string) error {
 	return c.client.Agent().CheckDeregister(ID)
 }
 
-// PeriodicSync triggers periodic syncing of services and checks with Consul.
-// This is a long lived go-routine which is stopped during shutdown
-func (c *ConsulService) PeriodicSync() {
-	sync := time.NewTicker(syncInterval)
+// Run triggers periodic syncing of services and checks with Consul.  This is
+// a long lived go-routine which is stopped during shutdown.
+func (c *Syncer) Run() {
+	d := initialSyncDelay + lib.RandomStagger(initialSyncBuffer-initialSyncDelay)
+	sync := time.NewTimer(d)
+	c.logger.Printf("[DEBUG] consul.sync: sleeping %v before first sync", d)
+
 	for {
 		select {
 		case <-sync.C:
+			d = syncInterval - lib.RandomStagger(syncInterval/syncJitter)
+			sync.Reset(d)
```

Comment:
Why not reset the timer once the sync completes? We might trigger syncs concurrently if the performSync takes longer than the syncInterval(which is unlikely, but possible if the http call hangs for some reason?)